modules/effects/src/effect_notification.ts

```diff
@@ -19,13 +19,16 @@ export function reportInvalidActions(
     const isInvalidAction = !isAction(action);
 
     if (isInvalidAction) {
-      reporter.handleError(
-        new Error(
-          `Effect ${getEffectName(
-            output
-          )} dispatched an invalid action: ${stringify(action)}`
-        )
-      );
+      try {
+        // handleError can throw errors
+        reporter.handleError(
+          new Error(
+            `Effect ${getEffectName(
+              output
+            )} dispatched an invalid action: ${stringify(action)}`
+          )
+        );
+      } catch (error) {}
```

Comment:
This just swallows the error. I think error handler should not be throwing errors to begin with.

Reply:
I think it is a common practice, that developers throw in errorHandler, to stop the js from executing. This is also the default behavior without having implemented a custom error handler:
https://medium.com/@amcdnl/global-error-handling-with-angular2-6b992bdfb59c

Reply:
I see. What if it's done unintentionally? Debugging swallowed exceptions would be really hard, right?

Reply:
Yes, we should log it at least.
We could do a `console.error` in the catch?

Reply:
That's what default error handler does: https://github.com/angular/angular/blob/0cd843169884a57d92c1ec114f7e778720b5ef24/packages/core/src/errors.ts#L30

Reply:
btw, the default `ErrorHandler` from angular does not throw error: https://github.com/angular/angular/blob/master/packages/core/src/error_handler.ts#L44

Reply:
Ok I see.

Reply:
The question is if this should still be handled in case error handler throws?

Reply:
I would not expect that the error handler throws an error.
If it's an unexpected exception, my opinion is that it must be handled in the error handler itself. But if we want to resubscribe on error, we can't make this assumption, and expect that it's possible that the error handler throws.

Reply:
Ok, I will close the PR. I just had some problems with this but I am not throwing in the error handler anymore.